flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/common/utils/GoogleCredentialsSupplier.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (C) 2023 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.flink.bigquery.common.utils;
+
+import org.apache.flink.annotation.Internal;
+
+import org.apache.flink.shaded.curator5.com.google.common.io.BaseEncoding;
+
+import com.google.auth.Credentials;
+import com.google.auth.oauth2.AccessToken;
+import com.google.auth.oauth2.GoogleCredentials;
+
+import java.io.ByteArrayInputStream;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.Optional;
+
+/** A utility class to supply credentials given the multiple possible configuration sources. */
+@Internal
+public class GoogleCredentialsSupplier {
```

Comment:
It looks like we could reuse `BigQueryCredentialsSupplier.java` from `spark-bigquery-connector/bigquery-connector-common`. Have we considered to add `bigquery-connector-common` as a dependency for this flink connector? This also applies to some implementations under pkg `services`  where we have similar functions in `bigquery-connector-common`.

Reply:
Yes, we had an offline conversation with @davidrabinowitz about reusing the `bigquery-connector-common` library in a subsequent PR. Sadly I didn't knew the library was available as a separated component from the Spark connector (didn't wanted to include the whole thing for Flink purposes since there are stuff that wouldn't be used increasing the deps complexity and overall jar size). 

I will tag you both in the new PR when completed. 

For the moment, the BigQueryServices interface and implementations (runtime and tests) are still necessary given the lack of emulator support for self contained tests (unitary and integration).

Reply:
LGTM